src/components/Select/Select.tsx

```diff
@@ -0,0 +1,203 @@
+import React, { Ref, FC, useState, useEffect } from 'react';
+
+// Styles:
+import { Dropdown, ToggleState } from '../Dropdown';
+import { Menu } from '../Menu';
+import { TextInput, TextInputWidth } from '../Inputs';
+import { SelectOption, SelectProps } from './Select.types';
+import { Pill, PillSize } from '../Pills';
+import { IconName } from '../Icon';
+import { mergeClasses } from '../../shared/utilities';
+
+import styles from './select.module.scss';
+
+const Options = (options: SelectOption[], onOptionChange: Function) => {
+    return (
+        <Menu
+            items={options}
+            onChange={(value) => {
+                const option = options.find((option) => option.value === value);
+                onOptionChange(option);
+            }}
+        />
+    );
+};
+
+export const Select: FC<SelectProps> = React.forwardRef(
+    ({ options = [], ...rest }, _ref: Ref<HTMLButtonElement>) => {
+        const {
+            defaultValue,
+            disabled,
+            clearable = false,
+            filterable = false,
+            multiple = false,
+            loadOptions,
+            isLoading = false,
+            classNames,
+            style,
+        } = rest;
+        const [visible, setVisibility] = useState(false);
+        const [defaultValueShown, setDefaultValueShown] = useState(false);
+        const [selectedOption, setSelectedOption] = useState<SelectOption>({
+            text: '',
+            value: '',
+        });
+        const [filteredOptions, setFilteredOptions] = useState(options);
+        const [searchQuery, setSearchQuery] = useState('');
+        const [selectedOptions, setSelectedOptions] = useState<SelectOption[]>(
+            []
+        );
+
+        useEffect(() => {
+            setFilteredOptions(options);
+        }, [JSON.stringify(options)]);
```

Comment:
This will be pretty inefficiant, can you come up with a better way to do that same

Reply:
this is now changed to other implementation